aiogram/aiogram
# PyPy support

# Description



As reported in #984, aiogram v3 may be compatible with PyPy, so we may support it officially by running all the tests on PyPy.

Closes #984.

## Type of change



- New feature (non-breaking change that adds functionality)

# How has this been tested?



- Locally by running tests on pypy
- With GitHub Actions

## Local Test Configuration
- Operating system: Arch Linux
- Python version: PyPy 7.3.9 (Python 3.9.12)

# Checklist



- [x] My changes are compatible with minimum requirements of the project

# Changes summary

```
- Add PyPy to test matrix
- Workaround Poetry installer not working on Windows with PyPy
- Don't install `fast` extras with PyPy
- Don't run coverage on PyPy as it slows down tests with timeout a lot
- Merge "Lint code" and "Check code style" steps
- Add and improve step names
- Remove useless `max-parallel` matrix config
```
